docs/lab_08.html

```diff
@@ -98,12 +98,50 @@ <h2>Goals</h2>
 <ul>
 	<li>Learn how to commit changes to the repository</li>
 </ul>
-<h2>Commit the change</h2>
+
+<h2>Committing using an editor</h2>
 <p>Ok, enough about staging.  Let&#8217;s commit what we have staged to the repository.</p>
 <p>Committing is the action of saving the changes to the repository.  It is like taking a snapshot of the changes.  You can always go back to this snapshot later.</p>
-<p>So commit now and check the status.</p>
+<p>Previously, when committing the <code>hello.js</code> file, you entered the command <code>git commit -m &ldquo;Your commit message&rdquo;</code>. The <code>-m</code> flag will accept the commit message inline, within the &ldquo;double quotes&rdquo;. If you enter the command <code>git commit</code> without the <code>-m</code> flag, your default terminal editor will open </p>
+<p>If you omit the <code>-m</code> flag from the <code>git commit</code> command, git will open your default terminal editor.</p> 
+<p>The editor is chosen from the following list (in priority order):</p>
+<ul>
+  <li>GIT_EDITOR environment variable</li>
+	<li>core.editor configuration setting</li>
+	<li><span class="caps">VISUAL</span> environment variable</li>
+	<li><span class="caps">EDITOR</span> environment variable</li>
+</ul>
+<p>I have the <span class="caps">EDITOR</span> variable set to <code>vim</code>. Here's an example of what that would look like:</p>
+<pre class="sample">|
+  # Please enter the commit message for your changes. Lines starting
+  # with '#' will be ignored, and an empty message aborts the commit.
+  # On branch master
+  # Changes to be committed:
+  #   (use "git reset HEAD &lt;file&gt;..." to unstage)
+  #
+  #	modified:   hello.js
+  #
+</pre>
+<p>On the first line, you would enter the commit message for example: &#8220;Using process.argv&#8221;.  Save the file and exit the editor. </p>
+<p>Here are some basic commands for vim:</p>
+<ul>
+  <li>To add a commit message, you would type <code>i</code> to enter insert mode, then type your message.</li>
+  <li>To save and exit the file in vim, you would type <code>:wq</code> and press enter.  The <code>:</code> puts vim in command mode, the <code>w</code> writes the file, and the <code>q</code> quits the editor.</li>
+  <li>To exit without saving, you would type <code>:q!</code> and press enter.  The <code>!</code> forces the quit without saving.</li>
+</ul>
```

Comment:
Maybe make a table for the shortcuts

```suggestion
          <li>
            To save and exit the file in vim, you would first press the
            <code>Escape</code> key, then type <code>:wq!</code> and press
            Enter.
          </li>
          <li>
            To exit without saving, you would type <code>:q!</code> and press
            enter. The <code>!</code> forces the quit without saving.
          </li>
        </ul>
        <table class="shortcuts">
          <thead>
            <tr>
              <th>Key</th>
              <th>Function</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td><code>Esc</code></td>
              <td>exits insert mode</td>
            </tr>
            <tr>
              <td><code>:</code></td>
              <td>puts vim in command mode</td>
            </tr>
            <tr>
              <td><code>w</code></td>
              <td>writes the file</td>
            </tr>
            <tr>
              <td><code>q</code></td>
              <td>quits the editor</td>
            </tr>
            <tr>
              <td><code>!</code></td>
              <td>forces the editor to quit</td>
            </tr>
          </tbody>
        </table>
```


here's some css that can be added to `docs/screen.css` for the table

```css
.shortcuts code {
  color: #74e83f;
  background: #2e2e2e;
}

.shortcuts {
  border-radius: 5px;
  padding: 8px;
  border: 1px solid #ddd;
}
.shortcuts th {
  text-align: left;
  border-bottom: 1px solid #ddd;
  margin-bottom: 4px;
}

.shortcuts td {
  padding: 4px 0;
  border-bottom: 1px solid #ddd;
}

.shortcuts td:nth-child(1),
.shortcuts th:nth-child(1) {
  padding-right: 8px;
  border-right: 1px solid #ddd;
  display: flex;
  justify-content: center;
}

.shortcuts td:nth-child(2),
.shortcuts th:nth-child(2) {
  padding-left: 8px;
}

.shortcuts td:nth-child(1) code {
  border-radius: 5px;
  color: #74e83f;
  background: #2e2e2e;
}

.shortcuts tr:last-child td {
  border-bottom: none;
  padding-bottom: 0;
}
```